Objects/boolobject.c

```diff
@@ -72,9 +72,9 @@ bool_vectorcall(PyObject *type, PyObject * const*args,
     assert(PyType_Check(type));
     if (nargs) {
         ok = PyObject_IsTrue(args[0]);
-    }
-    if (ok < 0) {
-        return NULL;
+        if (ok < 0) {
+            return NULL;
+        }
     }
     return PyBool_FromLong(ok);
```

Comment:
I assume it won't make a difference because `PyBool_FromLong` will be likely smashed (inlined :wink: ) by the compiler, but can you check if there is any improvement in making the branch yourself here?

Basically changing the `PyBool_FromLong(ok)` to a conditional with `Py_RETURN_TRUE` and `Py_RETURN_FALSE`

Reply:
@pablogsal 
I checked it locally and there was no improvement. :) 
I will merge this PR without changing ;)

Thank you for the review!

Reply:
@pablogsal If you want to ensure that a function is inlined, I suggest to add a new variant of PyBool_FromLong() which is delcared as a static inline function.